engine/src/parameters.rs

```diff
@@ -130,13 +130,42 @@ impl SubmitResult {
     }
 }
 
-/// Borsh-encoded parameters for the `call` function.
+/// Borsh-encoded parameters for the engine `call` function.
 #[derive(BorshSerialize, BorshDeserialize)]
 pub struct FunctionCallArgs {
     pub contract: RawAddress,
+    /// Wei compatible Borsh-encoded value field to attach an ETH balance to the transaction
+    pub value: WeiU256,
     pub input: Vec<u8>,
 }
 
+/// Legacy Borsh-encoded parameters for the engine `call` function, to provide backward type compatibility
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct FunctionCallArgsLegacy {
+    pub contract: RawAddress,
+    pub input: Vec<u8>,
+}
+
+/// Deserialized values from bytes to current or legacy Borsh-encoded parameters
+/// for passing to the engine `call` function, and to provide backward type compatibility
+#[derive(BorshSerialize, BorshDeserialize)]
+pub enum CallArgs {
+    New(FunctionCallArgs),
+    Legacy(FunctionCallArgsLegacy),
+}
+
+impl CallArgs {
+    pub fn deserialize(bytes: &[u8]) -> Option<Self> {
+        if let Ok(value) = FunctionCallArgs::try_from_slice(bytes) {
+            Some(Self::New(value))
```

Comment:
```suggestion
        if let Ok(value) = Self::try_from_slice(bytes) {
            Some(value)
```

Reply:
Addressed notes and remarks here - made a commit with improvements and corrections:
https://github.com/aurora-is-near/aurora-engine/pull/351#issuecomment-970794864